### DISCUSSION TITLE
How can I change cluster colors dynamically based on the max count of an attribute ?

### DISCUSSION THREAD

**Discussion by TheUncharted** [rnmapbox/maps#3095] [Q&A]
Hi

I have some posts with an attribute category 
I would like my cluster point to have the color of the category which has the most counts like this picture

![image](https://github.com/rnmapbox/maps/assets/24917492/0063f3fc-b89c-47d6-adca-e09db92c10cd)

Or if the max is not possible how can I display a custom marker like this example where the count are displayed
I found this picture on mapbox github issue 
[link](https://github.com/mapbox/mapbox-gl-js/issues/4491)

![image](https://github.com/rnmapbox/maps/assets/24917492/4524c44d-2cd9-4013-a790-94bdaea74ace)

**Comment by mfazekas** [comment]
@TheUncharted you should be able to extract the max with`clusterProperties`. Then you can use set the color based on that property.

The second one is not really possible. You might try using iconImage with an expression generated from like: '23-30-56-90' which would mean 23 in first, 30 in second etc category. Then you could use the onImageMissing to generate the proper image for that name.

**Reply by TheUncharted** [reply]
i tried something like that

```
clusterProperties={{
  category1: [
    ["+", ["accumulated"], ["case", ["==", ["get", "category"], "category1"], 1, 0]],
  ],
  category2: [
    ["+", ["accumulated"], ["case", ["==", ["get", "category"], "category2"], 1, 0]],
  ],
  category3: [
    ["+", ["accumulated"], ["case", ["==", ["get", "category"], "category3"], 1, 0]],
  ],
  MaxCategory: [
    ["max", ["get", "category1"], ["get", "category2"], ["get", "category3"]],
  ],
}}

```
and got this error if I add the max
I m not sure of the syntax

com.mapbox.maps.MapboxStyleException: Add source failed: GeoJSON source clusterProperties member must be an array with length of 2
  com.mapbox.maps.extension.style.sources.Source.bindTo(Source.kt:60)
  com.mapbox.maps.extension.style.sources.generated.GeoJsonSource.bindTo(GeoJsonSource.kt:107)
  com.mapbox.maps.extension.style.sources.SourceUtils.addSource(SourceExt.kt:59)
  com.mapbox.rctmgl.components.styles.sources.RCTSource.addToMap(RCTSource.kt:129)
  com.mapbox.rctmgl.components.styles.sources.RCTSource.addToMap(RCTSource.kt:155)
